Model/Config/Source/ThirdPartyPaymentVault.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Swarming\SubscribePro\Model\Config\Source;
+
+class ThirdPartyPaymentVault implements \Magento\Framework\Data\OptionSourceInterface
+{
+    /**
+     * @var array
+     */
+    private $supportedVaults;
+
+    /**
+     * @param array $supportedVaults
+     */
+    public function __construct(
+        array $supportedVaults = []
+    ) {
+        $this->supportedVaults = $supportedVaults;
+    }
+
+    /**
+     * @return array
+     */
+    public function toOptionArray(): array
+    {
+        $options = [];
+
+        foreach ($this->supportedVaults as $vault) {
+            $options[] = ['value' => $vault['code'], 'label' => $vault['name']];
+        }
+
+        return $options;
+    }
```

Comment:
I think the source model should display value payment methods that are installed and allowed.
We can take all installed payment methods and filter method code by allowed/supported codes that are passed in the constructor.

Reply:
done